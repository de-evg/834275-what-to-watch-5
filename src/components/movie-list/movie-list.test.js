import React from "react";
import renderer from "react-test-renderer";
import MovieList from "./movie-list";
import SmallMovieCard from "../small-movie-card/small-movie-card";
import SmallVideoPlayer from "../small-video-palyer/small-video-player";
import {BrowserRouter} from "react-router-dom";

const movieFirst = {
  id: 0,
  title: `The Grand Budapest`,
  genre: `Horror`,
  release: 2014,
  posterURL: `/img/the-grand-budapest-hotel-poster.jpg`,
  previewURL: `/img/bg-the-grand-budapest-hotel.jpg`,
  rating: `8,9`,
  ratingLevel: `Good`,
  ratingCount: 10,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  videoURL: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  isInWatchList: Math.random() <= 1,
  runTime: 146
};

const movieSecond = Object.assign(
    {},
    movieFirst,
    {id: 1}
);

const movies = [movieFirst, movieSecond];

const cardWidthProps = (Component) => {
  return <Component
    movie={movies[1]}
    onMouseOut={() => {}}
    onMouseOver={() => {}}
    removeTimeDelay={() => {}}
    id={movies[1].id}
    key={movies[1].id}
  />;
};

const playerWidthProps = (Component) => {
  return <Component
    movie={movies[0]}
    isPlaying={false}
    onMouseOut={() => {}}
    id={movies[0].id}
    key={movies[0].id}
  />;
};

it(`Should MovieList render correctly`, () => {
  const tree = renderer
    .create(
        <BrowserRouter>
          <MovieList
            movies={movies}
            activeMovieID={0}
            renderSmallMovieCard={() => cardWidthProps(SmallMovieCard)}
            renderSmallVideoPlayer={() => playerWidthProps(SmallVideoPlayer)}
          />
        </BrowserRouter>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
