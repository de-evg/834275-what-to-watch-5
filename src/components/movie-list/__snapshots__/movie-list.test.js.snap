// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MovieList render correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    id={0}
  >
    <a
      className="small-movie-card__image"
      href="/films/0"
      onClick={[Function]}
    >
      <video
        autoPlay={false}
        className="small-movie-card__image"
        muted={true}
        onMouseOut={[Function]}
        poster="/img/bg-the-grand-budapest-hotel.jpg"
      >
        <source
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
          type="video/mp4"
        />
      </video>
    </a>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    id={1}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <a
        className="small-movie-card__image"
        href="/films/1"
        onClick={[Function]}
        style={
          Object {
            "display": "block",
            "zIndex": 4,
          }
        }
      >
        <img
          alt="The Grand Budapest"
          height="175"
          src="/img/bg-the-grand-budapest-hotel.jpg"
          width="280"
        />
      </a>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/1"
        onClick={[Function]}
      >
        The Grand Budapest
      </a>
    </h3>
  </article>
</div>
`;
